packages/next/src/routes/rest/buildFormState.ts

```diff
@@ -23,8 +23,32 @@ export const getFieldSchemaMap = (config: SanitizedConfig): FieldSchemaMap => {
 
 export const buildFormState = async ({ req }: { req: PayloadRequest }) => {
   const reqData: BuildFormStateArgs = req.data as BuildFormStateArgs
-
-  // TODO: run ADMIN access control for user
+  const userCollectionSlug = req.user?.collection
+
+  // If we have a user slug, test it against the functions
+  if (userCollectionSlug) {
+    const adminAccessFunction = req.payload.collections[userCollectionSlug].config.access?.admin
+
+    // Run the admin access function from the config if it exists
+    if (adminAccessFunction) {
+      const canAccessAdmin = await adminAccessFunction(req)
+
+      if (!canAccessAdmin) {
+        return Response.json(null, {
+          status: httpStatus.UNAUTHORIZED,
+        })
+      }
+      // Match the user collection to the global admin config
+    } else if (req.payload.config.admin.user !== userCollectionSlug) {
```

Comment:
```suggestion
    } else if (adminUserSlug !== incomingUserSlug) {
```

Reply:
Another comment here: since you're checking `payload.config.admin.user` off the `req`, isn't this a security vulnerability? An attacker could potentially just send a passing config. I _think_ we need to import the config directly here.